Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabFocusHandler.cpp

```diff
@@ -273,31 +284,60 @@ namespace AzToolsFramework::Prefab
         // Refresh the read-only cache, if the interface is initialized.
         if (m_readOnlyEntityQueryInterface)
         {
-            EntityIdList containerEntities;
+            EntityIdList entities;
 
             if (previousFocusedInstance.has_value())
             {
-                containerEntities.push_back(previousFocusedInstance->get().GetContainerEntityId());
+                previousFocusedInstance->get().GetEntities(
+                    [&](AZStd::unique_ptr<AZ::Entity>& entity) -> bool
+                {
+                    entities.push_back(entity->GetId());
+                    return true;
+                }
+                );
+                entities.push_back(previousFocusedInstance->get().GetContainerEntityId());
             }
-            containerEntities.push_back(focusedInstance->get().GetContainerEntityId());
+            entities.push_back(focusedInstance->get().GetContainerEntityId());
+            focusedInstance->get().GetEntities(
+                [&](AZStd::unique_ptr<AZ::Entity>& entity) -> bool
+            {
+                entities.push_back(entity->GetId());
+                return true;
+            });
 
-            m_readOnlyEntityQueryInterface->RefreshReadOnlyState(containerEntities);
+            m_readOnlyEntityQueryInterface->RefreshReadOnlyState(entities);
         }
 
         // Refresh path variables.
         RefreshInstanceFocusPath();
 
         // Open all container entities in the new path.
         SetInstanceContainersOpenState(m_rootAliasFocusPath, true);
+        // Set open state on all nested instances in the new focus subtree based on edit scope.
+        SetInstanceContainersOpenStateOfAllDescendantContainers(
+            GetInstanceReference(m_rootAliasFocusPath), m_prefabEditScope == PrefabEditScope::SHOW_NESTED_INSTANCES_CONTENT);
 
-        PrefabFocusNotificationBus::Broadcast(&PrefabFocusNotifications::OnPrefabFocusChanged);
+        AZ::EntityId previousFocusedInstanceContainerEntityId = previousFocusedInstance.has_value() ?
+            previousFocusedInstance->get().GetContainerEntityId() : AZ::EntityId();
+        AZ::EntityId currentFocusedInstanceContainerEntityId = focusedInstance.has_value() ?
+            focusedInstance->get().GetContainerEntityId() : AZ::EntityId();
+        if (previousFocusedInstanceContainerEntityId != currentFocusedInstanceContainerEntityId)
+        {
+            PrefabFocusNotificationBus::Broadcast(&PrefabFocusNotifications::OnPrefabFocusChanged,
+                previousFocusedInstanceContainerEntityId, currentFocusedInstanceContainerEntityId);
+        }
 
         // Force propagation on both the previous and the new focused instances to ensure they are represented correctly.
+        // The most common operation is focusing a prefab instance nested in the currently focused instance.
+        // Queuing the previous focus before the new one saves some time in the propagation loop on average.
```

Comment:
How does it save time by queuing the previously focused instance?

Reply:
I'm not sure if this is still the case with the new deserialization changes, but at the time of writing putting the instance higher up in the hierarchy before the ones deeper down would mean that the latter would be already taken care by the former, skipping one propagation step.
(Definitely a very small optimization, but still).